src/server/app/main.py

```diff
@@ -0,0 +1,150 @@
+from datetime import datetime
+from typing import Annotated, List, Literal, Optional
+
+from fastapi import FastAPI, HTTPException, Query
+from pydantic import PositiveInt
+from stac_fastapi.types.rfc3339 import str_to_interval
+from stac_pydantic.shared import BBox
+
+from app.catalog_collection_search import CatalogCollectionSearch, DatetimeInterval
+from app.catalog_search_service import CatalogSearchService
+from app.cmr_collection_search import CMRCollectionSearch
+from app.config import Settings
+from app.models import SearchResponse
+from app.stac_api_collection_search import STACAPICollectionSearch
+
+invalid_bbox = HTTPException(status_code=400, detail="Invalid bbox")
```

Comment:
You might want to convert this to a function so you can pass in the invalid value and add it to the error message so the user can see what is actually invalid. One of my pet peeves are error messages indicating something is invalid, without indicating the invalid value, making it generally infuriating to try to track down the invalid value.